client.go

```diff
@@ -78,9 +83,10 @@ type APIDetails struct {
 // occurs. This includes messages that should hopefully provide useful context
 // to the end user.
 type APIErrorObject struct {
-	Code    int      `json:"code,omitempty"`
-	Message string   `json:"message,omitempty"`
-	Errors  []string `json:"errors,omitempty"`
+	Code           int      `json:"code,omitempty"`
+	Message        string   `json:"message,omitempty"`
+	Errors         []string `json:"errors,omitempty"`
+	RequiredScopes string   `json:"required_scopes,omitempty"`
```

Comment:
This still feels a little weird to me since it is not part of our standard API Error schema. Let me follow up on this one and get back to you.

Reply:
Let's handle required scopes through `errors` array as the standard way.